src/app/component/PushNotification/PushNotification.container.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { NotificationType } from 'Type/NotificationList';
+import { setBrowserPermission } from 'Store/Notification';
+import PushNotification from './PushNotification.component';
+import defaultImage from '../../../../media/preview.png';
+
+const mapStateToProps = state => ({
+    supported: state.NotificationReducer.nativeSupported,
+    grantType: state.NotificationReducer.nativeGrantType
+});
+
+const mapDispatchToProps = dispatch => ({
+    setPermissions: (value) => {
+        dispatch(setBrowserPermission(value));
+    }
+});
+
+const MappedPushNotification = connect(mapStateToProps, mapDispatchToProps)(PushNotification);
+
+const PushNotificationContainer = ({ notification }) => {
```

Comment:
Very interesting. We do not have anything like this around in the project. What is the main purpose bof this?

How could we integrate this into the application? Will this help use somehow? Could you please investigate an improvement in this term? 

Contact me if not clear what I am asking for. Essentially, I would like to request an application audit, maybe you can advice some improvements to overall application quality / logic. There must be a lot of outdated solutions, which are hard to spot, because we are already "blurred" our vision.

Reply:
My initial goal for this was to keep PushNotification.component as it is, because it pretty much fully covers web notifications API. However, we can't pass all the props to it from the NotificationList.component without rewriting logic for existing notifications, so my idea was to make an interface, that parses data we get in NotificationList.component so that PushNotification can understand that without changing the core of the component itself, because we (or people using ScandiPWA later) might want to use full functionality present in the component. And finally I needed to connect to Redux, so I decided to map that into variable.

As for possible integrations, it's probably a very rare case and it was mostly used as a work-around so that I don't have to rewrite Redux store and components for Notifications, because that could lead to some more unwanted bugs.

Not sure if you also asked about line 21, but that's stateless functional component and they can be used in any component without state. I think I saw some places where we can implement that. As far as I know, they are little bit faster than class components and are also stupidly simple to understand even for those not familiar with React.

I will take a look at existing code and maybe give some advice!